components/script/script_thread.rs

```diff
@@ -377,14 +437,19 @@ impl Documents {
 
 #[allow(unrooted_must_root)]
 pub struct DocumentsIter<'a> {
-    iter: hash_map::Iter<'a, PipelineId, JS<Document>>,
+    iter: hash_map::Iter<'a, PipelineId, DocumentState>,
 }
 
 impl<'a> Iterator for DocumentsIter<'a> {
     type Item = (PipelineId, Root<Document>);
 
     fn next(&mut self) -> Option<(PipelineId, Root<Document>)> {
-        self.iter.next().map(|(id, doc)| (*id, Root::from_ref(&**doc)))
+        while let Some((id, state)) = self.iter.next() {
```

Comment:
I don't think it can, the important part is that the iterator is recomputed each time round the loop.